### DISCUSSION TITLE
`keyword1` and `keyword2` have the same color?

### DISCUSSION THREAD

**Discussion by ghost** [SpartanJ/ecode#217] [General]
I have tried different color schemes. I found that `keyword1` and `keyword2` always have the same color. `keyword3` is in different color. And if you set a keyword as `keyword4` then it will become invisible. Please correct me if I'm wrong.

**Comment by SpartanJ** [comment]
Not always, the default color scheme does use different colors. I don't remember by memory each case but it's true that sometimes it's repeated. And there's not a particular rule on when to use keyword and keyword2, but keyword3 is always used for function parameters. You'll be required to have  semantic-highlighting working with some LSP to be able to see it in use (any good LSP provides it, clang, or rust-analyzer or typescript-language-server). As for keyword4, it does not exist, editor currently supports 3 keywords, adding more is trivial but you need to add that color to each color-scheme and it's not that easy because not many support that many colors (currently most popular color-schemes provide a color palette that you need to follow, and the number of colors is limited).
